### ISSUE TITLE
Linear components table

### ISSUE THREAD

**Issue by famiu** [famiu/feline.nvim#145]
I'm thinking about making the components table linear (no more sections) and instead make the section gaps a provider, so everytime you want a new section, you'd have a component with that provider inbetween, and use that to set the highlights for the section gap. That'd be killing two birds with one stone since it'd have default background for section gaps by default, and it'd also allow you to customize the section gap. It'd also allow simplifying the truncation code by a great amount.

The issue is, it'd be a breaking change, and perhaps be a bit less intuitive than the current format. I'd like to hear people's thoughts on this though before I go about implementing it

**Comment by jonatan-branting** [comment]
I think this is a step in the right direction, making the configuration more general while also reducing complexity. Better do it now, as we're not at 1.0 yet :)

**Comment by EdenEast** [comment]
I agree with this direction. This makes the section gaps explicit. This area was ambiguous when I was constructing my status line. I also agree that if you were to make the change it is better when feline is still in v0.
